flake.nix

```diff
@@ -435,6 +436,26 @@
               platforms = platforms.all;
             };
           };
+          postgresql_17_src = pkgs.stdenv.mkDerivation {
+            pname = "postgresql-17-src";
+            version = postgresql_17.version;
+            src = postgresql_17.src;
+
+            nativeBuildInputs = [ pkgs.bzip2 ];
+
+            phases = [ "unpackPhase" "installPhase" ];
+
+            installPhase = ''
+              mkdir -p $out
+              cp -r . $out
+            '';
+            meta = with pkgs.lib; {
+              description = "PostgreSQL 17 source files";
+              homepage = "https://www.postgresql.org/";
+              license = licenses.postgresql;
+              platforms = platforms.all;
+            };
+          };
```

Comment:
`flake.nix` has a lot already and it's getting hard to maintain, could we move some derivations to the `nix` directory?

Reply:
@steve-chavez sure go for it

Reply:
We can definitely refactor this, but going to be a follow up PR